src/components/Search/SearchListWithHeader.tsx

```diff
@@ -98,23 +105,27 @@ function SearchListWithHeader(
                 return;
             }
 
-            if (item.transactions.every((transaction) => selectedItems[transaction.keyForList]?.isSelected)) {
-                const reducedSelectedItems: SelectedTransactions = {...selectedItems};
+            if (item.transactions.every((transaction) => selectedTransactions[transaction.keyForList]?.isSelected)) {
+                const reducedSelectedTransactions: SelectedTransactions = {...selectedTransactions};
+                setSelectedReports((prevReports) => prevReports.filter((reportID) => reportID !== item.reportID));
 
                 item.transactions.forEach((transaction) => {
-                    delete reducedSelectedItems[transaction.keyForList];
+                    delete reducedSelectedTransactions[transaction.keyForList];
                 });
 
-                setSelectedItems(reducedSelectedItems);
+                setSelectedTransactions(reducedSelectedTransactions);
                 return;
             }
 
-            setSelectedItems({
-                ...selectedItems,
+            if (item.reportID) {
+                setSelectedReports([...selectedReports, item.reportID]);
+            }
```

Comment:
I think this is true for both transactions and reports, since both have `reportID`. Do we want to select the report when selecting the transaction too? If so, I think we can remove the conditional, if not we can use something like `!item.transactionID`

Reply:
We are handling the `TransactionListItemType` on line 92, this is just a safe measure since `reportID` is optional on the `ListItem` type.